emcc.py

```diff
@@ -1111,6 +1111,9 @@ def check(input_file):
       separate_asm = True
       shared.Settings.FINALIZE_ASM_JS = False
 
+    if shared.Settings.GLOBAL_BASE < 0:
+      shared.Settings.GLOBAL_BASE = 8 # default if nothing else sets it
```

Comment:
No, it's a little more complex. In some modes, specifically binaryen and emterpreter, we need high values of this. Knowing if the user overrode it is helpful for the error message.